src/cloudevent/partials/pubsub/pubsub-on-message-published.ts

```diff
@@ -0,0 +1,39 @@
+import {DeepPartial, MockCloudEventPartials} from '../../types';
+import {MessagePublishedData} from 'firebase-functions/lib/v2/providers/pubsub';
+import {CloudEvent, CloudFunction} from 'firebase-functions/lib/v2';
+import {getEventFilters, getEventType, PROJECT_ID} from '../helpers';
+
+export const pubsubOnMessagePublished:
+  MockCloudEventPartials<MessagePublishedData, MessagePublishedData> = {
+  generatePartial(cloudFunction: CloudFunction<MessagePublishedData>): DeepPartial<CloudEvent<MessagePublishedData>> {
+    const topicId = getEventFilters(cloudFunction)?.topic || '';
+    const source = `//pubsub.googleapis.com/projects/${PROJECT_ID}/topics/${topicId}`;
+    const subscription = `projects/${PROJECT_ID}/subscriptions/pubsubexample-1`;
+
+    const dataMessageJson = '{"hello": "world"}';
```

Comment:
1. This should be an actual object
2. I had assumed we'd make sure that `json` and `data` would be kept in sync. So if a user passed `json` in their partial we'd base 64 encode that as `data`. But that would require rearchitecting the design so you always take a partial in the generate event request

Reply:
> But that would require rearchitecting the design so you always take a partial in the generate event request

That exact part I forgot to take into account.

Only "solution" I landed at was to add an optional `postPartialGeneration` that takes in the generated data combined with the user provided data.

Not sure if that makes things messier or not.